lib/IDE/Refactoring.cpp

```diff
@@ -6538,30 +6588,59 @@ class AsyncConverter : private SourceEntityWalker {
   void addFuncDecl(const FuncDecl *FD) {
     auto *Params = FD->getParameters();
     auto *HandlerParam = TopHandler.getHandlerParam();
+    auto ParamPos = TopHandler.handlerParamPosition();
 
     // If the completion handler parameter has a default argument, the async
     // version is effectively @discardableResult, as not all the callers care
     // about receiving the completion call.
     if (HandlerParam && HandlerParam->isDefaultArgument())
       OS << tok::at_sign << "discardableResult" << "\n";
 
-    // First chunk: start -> the parameter to remove (if any)
-    SourceLoc LeftEndLoc = Params->getLParenLoc().getAdvancedLoc(1);
-    if (TopHandler.Index - 1 >= 0) {
+    // First chunk: start -> the parameter to remove (if any) +- comma
```

Comment:
I can remove that part if you want... The "handling of commas" is the fact that `Middle` uses `Params->get(TopHandler.Index)->getStartLoc()` rather than what `Last` does. Ie. `Middle` includes the comma where as `Last` does not. Perhaps it's clear enough now that I've separated each case properly.

Reply:
I'll comment in each case instead.